### PR TITLE
MIME types little refactoring

### PR DESCRIPTION
**Pull request by Sami32**
---
This change is [Reviewable](https://reviewable.io/reviews/digitalmediaserver/digitalmediaserver/89)

### CODE REVIEW

**Comment by Sami32** [comment]
I can remove all the images MIME types if you can confirm me that these code lines will not be need for a later UPnP implementation.

**Comment by Sami32** [comment]
If you meant MPEGTS, no, only a bad hardcoded hack for Sony AFAIK.

**Comment by Sami32** [comment]
`dnxhd` iss used by the old MI versions:
https://github.com/MediaArea/MediaInfoLib/commit/2fe6120c9c77b8c872477a843f703b1b59630d14

**Comment by Sami32** [comment]
I can see it in the commit under the draft, weird.

**Comment by Nadahar** [comment]
I've added it when I resolved conflicts between my local copy of the branch and the remote one. That means that it was there earlier. My question was why you removed it.

**Comment by Sami32** [comment]
No idea, it is too far now

**Comment by Sami32** [comment]
No i don't see any wrong could happen.
When no container could be determined the audio codec is applied.
When the mime test after the container cases switches is null then
```
if (mimeType == null) {
			mimeType = HTTPResource.getDefaultMimeType(getMediaType());
}
```
So in any cas a default MIME Type will be applied, as a container, only if FFmpeg or MI can identify it of course.

**Comment by Nadahar** [comment]
> When no container could be determined the audio codec is applied.

Where is this code? For me it seems like you removed the code that did this. Did you move it somewhere else?

**Comment by Sami32** [comment]
I saw it line 1656

**Comment by Nadahar** [comment]
@Sami32 You've got to link to the code you mean. If you mean `HTTPResource.getDefaultMimeType()`, that doesn't consider the codec at all. It only looks at the "type" (audio, video or image) and returns a "generic" MIME type for that "type". The only thing you can be certain of with this method, is that the returned MIME type is wrong.

**Comment by Sami32** [comment]
I was not able to find any new issue in the FFmpeg parsing in my 4 tests.

But keep it in standby i'll do some more tests later.

**Comment by Sami32** [comment]
I wasn't able to find any issues until now.

**Comment by Nadahar** [comment]
Sounds good :+1: Let me do some testing when I can catch up with myself :innocent:

File: src/main/java/net/pms/dlna/LibMediaInfoParser.java [hunk 1]
```diff
@@ -655,6 +661,10 @@ private static void getFormat(
 			value.equals("wmva")
 		) {
 			format = FormatConfiguration.VC1;
+		} else if (value.equals("vc-3")) {
```
Comment (Nadahar) [inline src/main/java/net/pms/dlna/LibMediaInfoParser.java]:
```suggestion
		} else if (value.equals("vc-3") || value.startsWith("dnxhd")) {
```

**Pull request merged**